lib/mapObjects/CBank.cpp

```diff
@@ -95,124 +95,130 @@ bool CBank::wasVisited (PlayerColor player) const
 	return !bc; //FIXME: player A should not know about visit done by player B
 }
 
-void CBank::onHeroVisit (const CGHeroInstance * h) const
+void CBank::onHeroVisit(const CGHeroInstance * h) const
 {
+	int banktext = 0;
+	ui16 soundID = soundBase::ROGUE;
+	switch (ID)
+	{
+	case Obj::DERELICT_SHIP:
+		banktext = 41;
+		break;
+	case Obj::DRAGON_UTOPIA:
+		banktext = 47;
+		break;
+	case Obj::CRYPT:
+		banktext = 119;
+		break;
+	case Obj::SHIPWRECK:
+		banktext = 122;
+		break;
+	case Obj::PYRAMID:
+		soundID = soundBase::MYSTERY;
+		banktext = 105;
+		break;
+	case Obj::CREATURE_BANK:
+	default:
+		banktext = 32;
+		break;
+	}
+	BlockingDialog bd(true, false);
+	bd.player = h->getOwner();
+	bd.soundID = soundID;
+	bd.text.addTxt(MetaString::ADVOB_TXT, banktext);
+	if (ID == Obj::CREATURE_BANK)
+		bd.text.addReplacement(getObjectName());
```

Comment:
default case also use text 32 and should use object name replacement

Reply:
fixed